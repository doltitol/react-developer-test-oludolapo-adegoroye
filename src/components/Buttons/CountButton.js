import React, { PureComponent } from 'react';

export class CountButton extends PureComponent {
  render() {
    return (
      <button
        onClick={this.props.onClick}
        className='count-button'
        style={{
          width: this.props.small ? '24px' : '45px',
          height: this.props.small ? '24px' : '45px',
          fontSize: this.props.small ? '20px' : '40px',
        }}
      >
        {this.props.text}
      </button>
    );
  }
}

export default CountButton;
