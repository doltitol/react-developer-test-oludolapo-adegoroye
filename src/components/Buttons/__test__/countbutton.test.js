import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CountButton from '../CountButton';

describe('<CountButton /> spec', () => {
  it('renders the component', () => {
    const container = render(<CountButton text='+' />);
    const getByTestId = container.getByTestId;
    const countButton = getByTestId('count-button');
    expect(countButton.textContent).toEqual('+');
    expect(countButton).toHaveClass('count-button');
    expect(countButton).toHaveStyle('fontSize: 40px');
    expect(container.children).toMatchSnapshot();
  });
});
