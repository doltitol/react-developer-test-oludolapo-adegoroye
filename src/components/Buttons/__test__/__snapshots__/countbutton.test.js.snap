// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountButton /> spec renders the component 1`] = `undefined`;
